HACKING

```diff
@@ -219,6 +219,10 @@ Coding
   ``gint``, use a ``gchar`` for individual (ASCII/UTF-8) string
   characters rather than ``gint``, and use a ``guint`` for integers
   which cannot be negative rather than ``gint``.
+* Avoid using untyped pointers (e.g. gpointer) where practical.
+* Prefer loops to calling *_foreach() with a non-NULL untyped 
+  ``user_data`` pointer parameter, unless external iteration is not 
+  supported (e.g. for tree structures).
```

Comment:
This is a bit confusing IMO. Maybe it would read better by just dropping the `with a non-NULL untyped user_data pointer parameter`? Even without that part, it's still a useful guideline and not overly specific.

Reply:
I've changed it to just "calling ``some_type_foreach()`` with a ``user_data`` argument". I think the last part implies that a NULL argument is OK. I agree that loops are generally better than *_foreach functions, but there are some cases where they're fine.

Reply:
> ...but there are some cases where they're fine

especially in languages with function expressions/lambdas :)